### ISSUE TITLE
Full Implementation Available + Disclaimer

### ISSUE THREAD

**Issue by OverLordGoldDragon** [GLambard/AdamW_Keras#2]
A complete implementation of AdamW, NadamW, and SGDW - with warm restarts, cosine annealing, and per-layer learning rate multipliers is available [here](https://github.com/OverLordGoldDragon/keras-adamw).

Regarding this repository's implementation, it has a major flaw: weight decay can only be set _globally_, for _all_ layer weight matrices; this is rarely useful, as not all weights require decay, and most require weight-specific treatment. Also, no mention is made against using both weight decay _and_ l2 penalty, which is important - as using both defeats the purpose of either.

Feedback is welcome.
